src/gamma_inc.jl

```diff
@@ -143,6 +155,230 @@ function rgammax(a::Float64,x::Float64)
+    else
+        tx = 2*x
+        r = a[n]
+        h = a[n-1] + r*tx
+        for k = n-2:-1:1
+            s=r
+            r=h
+            h=a[k]+r*tx-s
+        end
+        return a[1]/2.0 - r + h*x
+    end
+end
+"""
+    stirling(x)
+
+Compute the Stirling Series for given `x`.
+"""
+function stirling(x::Float64)
+  
+    a = zeros(18) 
+    if x < floatmin(Float64)*1000.0
+        return floatmax(Float64)/1000.0
+    elseif x < 1
+        return loggamma1p(x) - (x+0.5)*log(x)+x - log(sqrt(2*pi))
+    elseif x < 2
+        return loggamma1p(x-1) - (x-0.5)*log(x)+x - log(sqrt(2*pi))
+    elseif x < 3
+        return loggamma1p(x-2) - (x-0.5*log(x)+x  - log(sqrt(2*pi))+log(x-1))
+    elseif x < 12
+        a[1]=1.996379051590076518221
+        a[2]=-0.17971032528832887213e-2
+        a[3]=0.131292857963846713e-4
+        a[4]=-0.2340875228178749e-6
+        a[5]=0.72291210671127e-8
+        a[6]=-0.3280997607821e-9
+        a[7]=0.198750709010e-10
+        a[8]=-0.15092141830e-11
+        a[9]=0.1375340084e-12
+        a[10]=-0.145728923e-13
+        a[11]=0.17532367e-14
+        a[12]=-0.2351465e-15
+        a[13]=0.346551e-16
+        a[14]=-0.55471e-17
+        a[15]=0.9548e-18
+        a[16]=-0.1748e-18
+        a[17]=0.332e-19
+        a[18]=-0.58e-20
+        z=18.0/(x*x)-1.0
+        return chepolsum(z,a)/(12.0*x)
+    else
+        c = zeros(7)
+        z = 1.0/(x*x)
+        if x < 1000
+            return @horner(z, 0.25721014990011306473e-1, 0.82475966166999631057e-1, -0.25328157302663562668e-2, 0.60992926669463371e-3, -0.33543297638406e-3, 0.250505279903e-3)/((0.30865217988013567769 + z)/x)
+        else
+            return (((-z/1680.0+1.0/1260.0)*z-1.0/360.0)*z+1.0/12.0)/x
+        end
+    end
+end
+"""
+    gammax(x)
+
+gammax(x) = ``e^{stirling(x)}`` if x>0
+else ``\\Gamma(x)/(e^{-x + (x-0.5)*log(x)}/\\sqrt{2 \\pi}``
+"""
+function gammax(x::Float64)
+    if x>=3
+        return exp(stirling(x))
+    elseif x>0
+        return gamma(x)/(exp(-x+(x-0.5)*log(x))*sqrt(2*pi))
+    else
+        return floatmax(Float64)/1000.0
+    end
+end
+"""
+    lambdaeta(eta)
+
+Function to compute the value of eta satisfying ``eta^{2}/2 = \\lambda-1-log(\\lambda)``
+"""
+function lambdaeta(eta::Float64)
+  
+    ak = zeros(7)
+    s = eta*eta*0.5
+    if eta == 0.0
+        la = 1
+    elseif eta < -1.0
+        r = exp(-1-s)
+        la = @horner(r, 0.0, 1.0, 1.0, 1.5, 8.0/3.0, 125.0/24.0, 15.0/5.0)
+    elseif eta < 1.0
+        r = eta
+        la = @horner(r, 1.0, 1.0, 1.0/3.0, 1.0/36.0, -1.0/270.0, 1.0/4320.0, 1.0/17010.0)
+    else
+        r = 11 + s
+        l = log(r)
+        la = r+l
+        r = 1.0/r
+        l2=l*l
+        l3=l2*l
+        l4=l3*l
+        l5=l4*l
+        ak[1] = 1.0
+        ak[2] = (2 - l)*0.5
+        ak[3] = (-9*l+6+2*l2)/6.0
+        ak[4] = -(3*l3+36*l-22*l2-12)/12.0
+        ak[5] = (60 + 350*l2 - 300*l -125*l3 + 12*l4)/60.0
+        ak[6] = -(-120-274*l4+900*l-1700*l2+1125*l3+20*l5)/120.0
```

Comment:
Won't it look but clumsy?

Reply:
You can put each element on a different line.

Otherwise just use variables `ak1` etc: Julia isn’t quite as good as fortran as getting rid of unnecessary array allocations.